src/client.ts

```diff
@@ -808,6 +808,8 @@ export class StreamChat<StreamChatGenerics extends ExtendableGenerics = DefaultG
     this.activeChannels = {};
     // reset client state
     this.state = new ClientState();
+    // reset thread manager
+    this.threads = new ThreadManager({ client: this });
```

Comment:
This feels "leaky" and might cause issues because we're mutating the manager in place, appropriate components might not register this change until next re-render. The active threads within will stay connected to the dispatcher and will keep listening and updating their states.

I'd rather do something like: 
```suggestion
    this.threads.state.next(INITIAL_THREAD_MANAGER_STATE) // default would need to be created
    // or with new API
    this.threads.resetLocalState(); // or initializeLocalState() - not sure about naming, I'd let Matvei in on this, he's good at these kinds of things :)
```